app/core/agents/interpreter/tool_interpreter.py

```
@@ -37,112 +63,361 @@ class Interpreter(BaseTool):
+        }
+
+        with tempfile.NamedTemporaryFile("w", suffix=".json", delete=False) as payload_file:
+            json.dump(payload, payload_file)
+            payload_path = Path(payload_file.name)
+
+        with tempfile.NamedTemporaryFile("w", suffix=".json", delete=False) as result_file:
+            result_path = Path(result_file.name)
+
+        process = None
+        try:
+            process = subprocess.Popen(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)